tests/test_simulation.py

```diff
@@ -80,16 +81,18 @@ def test_no_network_config():
         with edit_config(config_path) as config:
             config.pop("network")
 
+        os.remove(config_path.parent / "circuit_config.json")
         simulation = test_module.Simulation(config_path)
 
-        with pytest.raises(BluepySnapError, match="No 'network' set"):
+        with pytest.raises(BluepySnapError, match="'network' file not found"):
             simulation.circuit
 
 
 def test_no_node_set():
     with copy_test_data(config="simulation_config.json") as (_, config_path):
         with edit_config(config_path) as config:
             config.pop("node_sets_file")
+            os.remove(config_path.parent / "circuit_config.json")
```

Comment:
Why the file `circuit_config.json` needs to be removed to make the test succeed?
Isn't it related to the configuration only?

Reply:
That is, again, due to [`libsonata`](https://github.com/BlueBrain/libsonata/blob/master/src/config.cpp#L944): If `node_sets_file` is not defined in the simulation config, `libsonata` fetches it from the circuit config. 

Technically, the correct approach would be to remove it also from the circuit config but this was just simpler. I'll clarify it with a comment.